arangod/Indexes/IndexFactory.cpp

```diff
@@ -220,6 +219,29 @@ Result IndexFactory::enhanceIndexDefinition( // normalizze deefinition
                      arangodb::velocypack::Value(std::to_string(id)));
     }
 
+    auto nameSlice = definition.get(StaticStrings::IndexName);
+    std::string name;
+
+    if (nameSlice.isString() && (nameSlice.getStringLength() != 0)) {
+      name = nameSlice.copyString();
+    } else {
+      // we should set the name for special types explicitly elsewhere, but just in case...
+      if (Index::type(type.copyString()) == Index::IndexType::TRI_IDX_TYPE_PRIMARY_INDEX) {
+        name = StaticStrings::IndexNamePrimary;
+      } else if (Index::type(type.copyString()) == Index::IndexType::TRI_IDX_TYPE_EDGE_INDEX) {
+        name = StaticStrings::IndexNameEdge;
```

Comment:
Will this lead to two edge indexes (in the RocksDB engine) end up with the same name? Is this going to be a problem?